pkgs/applications/editors/emacs/build-support/melpa.nix

```diff
@@ -62,17 +64,17 @@ in
     recipe: Optional MELPA recipe.
     Default: a minimally functional recipe
   */
-, recipe ? (writeText "${pname}-recipe" ''
-    (${ename} :fetcher git :url ""
-              ${lib.optionalString (files != null) ":files ${files}"})
+, recipe ? (writeText "${finalAttrs.pname}-recipe" ''
+    (${finalAttrs.ename} :fetcher git :url ""
+              ${lib.optionalString (finalAttrs.files != null) ":files ${finalAttrs.files}"})
```

Comment:
there are no obvious way to infer `finalAttrs.*` comes from `files`, sadly.

Reply:
Yeah, this seems hard for a parser.  I guess some evaluation is need.  Maybe provide a way to disable the `nixf-tidy` check?